### PR TITLE
Dynamic changes

### PR DESCRIPTION
**Pull request by vcloarec**
some changes for dynamic drivers and minor fix for statistic with inactive face

### CODE REVIEW

File: mdal/mdal_driver_manager.cpp [hunk 1]
```diff
@@ -262,16 +262,44 @@ void MDAL::DriverManager::loadDynamicDrivers()
...
+  char *path = nullptr;
+  if ( requiredPathSize != 0 )
+  {
+    path = ( char * )malloc( requiredPathSize * sizeof( char ) );
+    getenv_s( &requiredPathSize, path, requiredPathSize, "PATH" );
+  }
+  OutputDebugString( path ); OutputDebugString( "\n" );
+#endif
+#endif
+
```
Comment (PeterPetrik) [inline mdal/mdal_driver_manager.cpp:281]: remove? you can see the list of loaded drivers by `mdalinfo --drivers`?

File: mdal/mdal_driver_manager.cpp [hunk 2]
```diff
@@ -262,16 +262,44 @@ void MDAL::DriverManager::loadDynamicDrivers()
...
+#if 0 // for debugging library loading in release mode under Windows
+#ifdef WIN32
+      OutputDebugString( "+++++++++++++++++++++++++++++  loading driver is success for:" );  OutputDebugString( libFile.c_str() ); OutputDebugString( "\n" );
+    }
+    else
+    {
+      OutputDebugString( "-----------------------------   loading driver fails for:" );  OutputDebugString( libFile.c_str() ); OutputDebugString( "\n" );
+#endif
+#endif
+    }
```
Comment (PeterPetrik) [inline mdal/mdal_driver_manager.cpp:302]: remove? you can see the list of loaded drivers by `mdalinfo --drivers`?

File: mdal/mdal_utils.cpp [hunk 3]
```diff
@@ -1041,8 +1052,34 @@ bool MDAL::Library::loadLibrary()
...
+      LPSTR messageBuffer = nullptr;
+      size_t size = FormatMessage( FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                                   NULL, errorMessageID, MAKELANGID( LANG_NEUTRAL, SUBLANG_DEFAULT ), ( LPSTR )&messageBuffer, 0, NULL );
+
+      std::string message( messageBuffer, size );
+    }
+    OutputDebugString( "------------------------------  code error:" );  OutputDebugString( std::to_string( errorMessageID ).c_str() ); OutputDebugString( "\n" );
+    OutputDebugString( "------------------------------  message:" );  OutputDebugString( message.c_str() ); OutputDebugString( "\n" );
+  }
+#endif
```
Comment (PeterPetrik) [inline mdal/mdal_utils.cpp:1078]: I will create a new function from this `void debug_win_loading` and ifdef it to `if defined(windows) and defined(debug)` or so.?

Comment (PeterPetrik) [reply mdal/mdal_utils.cpp:1078]: or just remove it?

File: tools/externaldriver/mdal_external_driver.h [hunk 4]
```diff
@@ -8,21 +8,21 @@
...
+#define MDAL_LIB_EXPORT __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
+#endif
+#endif
+#else
+#if __GNUC__ >= 4
+#define MDAL_LIB_EXPORT __attribute__ ((visibility ("default")))
+#else
+#define MDAL_LIB_EXPORT
+#endif
+#endif //_WIN32 || defined __CYGWIN__
```
Comment (PeterPetrik) [inline tools/externaldriver/mdal_external_driver.h:25]: was looking better before?

File: tools/externaldriver/mdal_external_driver.h [hunk 5]
```diff
@@ -132,6 +135,9 @@ MDAL_LIB_EXPORT bool MDAL_DRIVER_D_hasActiveFlagCapability( int meshId, int grou
 //! Returns the dataset active flags
 MDAL_LIB_EXPORT int MDAL_DRIVER_D_activeFlags( int meshId, int groupIndex, int datasetIndex, int indexStart, int count, int *buffer );
 
+//! Unload data store in memory (for driver that support lazy loading, data are unloaded after statistic calculation)
```
Comment (PeterPetrik) [inline tools/externaldriver/mdal_external_driver.h:138]: Unloads the dataset from memory

File: tools/externaldriver/mdal_external_driver.h (hunk 4 above)
Comment (vcloarec) [reply tools/externaldriver/mdal_external_driver.h:25]: yes, I will change once back to Linux, it is a visual studio config fo astyle I didn't find...

File: mdal/mdal_utils.cpp (hunk 3 above)
Comment (vcloarec) [reply mdal/mdal_utils.cpp:1078]: maybe remove all this stuff once the implementation will be ok. For now, it is convenient in run time when testing...

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**